src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -398,6 +401,14 @@ public void messageReceived(RecoveryCleanFilesRequest request, TransportChannel
                 Store.MetadataSnapshot sourceMetaData = request.sourceMetaSnapshot();
                 try {
                     store.cleanupAndVerify("recovery CleanFilesRequestHandler", sourceMetaData);
+                } catch (CorruptIndexException | IndexFormatTooNewException | IndexFormatTooOldException ex) {
+                    // this is a fatal exception at this stage.
+                    // this means we transferred files from the remote that have not be checksummed and they are
+                    // broken. We have to clean up this shard entirely, remove all files and bubble it up to the
+                    // source shard since this index might be broken there as well? The Source can handle this and checks
+                    // its content on disk if possible.
+                    store.deleteContent(); // clean up and delete all files
```

Comment:
The reason we delete the temp files is that when a new recovery starts, it will create a new folder and copy files over again and we could (were) filling up disks that way. This is not an issue here. I think we should be consistent in our policies. We chose to leave complete corrupted shards around in other places, so I think we should do the same here? I don't feel strongly about this, though. This was the gist of my question.